tools/resource-gen/main.go

```diff
@@ -418,34 +407,127 @@ func main() {
 		resources = append(resources, resourceInfo)
 	}
 
-	var generatorTemplate *template.Template
+	var generatorFn GeneratorFn
 
 	switch gen {
 	case "type":
-		generatorTemplate = ResourceTemplate
+		generatorFn = TemplateGeneratorFn(ResourceTemplate)
 	case "crd":
-		generatorTemplate = CustomResourceTemplate
+		generatorFn = TemplateGeneratorFn(CustomResourceTemplate)
+	case "openapi":
+		generatorFn = openApiGenerator
 	default:
 		log.Fatalf("%s is not a valid generator option\n", gen)
 	}
 
-	outBuf := bytes.Buffer{}
-	if err := generatorTemplate.Execute(&outBuf, struct {
-		Package   string
-		Resources []ResourceInfo
-	}{
-		Package:   pkg,
-		Resources: resources,
-	}); err != nil {
-		log.Fatalf("template error: %s", err)
-	}
-
-	out, err := format.Source(outBuf.Bytes())
+	err := generatorFn(pkg, resources)
 	if err != nil {
 		log.Fatalf("%s\n", err)
 	}
+}
 
-	if _, err := os.Stdout.Write(out); err != nil {
-		log.Fatalf("%s\n", err)
+func openApiGenerator(pkg string, resources []ResourceInfo) error {
+	protoTypeToType := map[string]reflect.Type{
+		"Mesh":        reflect.TypeOf(v1alpha1.Mesh{}),
+		"MeshGateway": reflect.TypeOf(v1alpha1.MeshGateway{}),
```

Comment:
Worth adding a comment to indicate that it's here that new types should be added if we want to generate openAPI for it